src/devices/imuBosch_BNO055/imuBosch_BNO055.cpp

```diff
@@ -436,9 +435,9 @@ void BoschIMU::run()
 
     // TODO: how to optimally protect only code filling the data vector?
 
-    mutex.wait();
-
-    data.zero();
+    // In order to avoid zeros when a single read from a sensor is missing,
+    // initialize the new measure to be equal to the previous one
+    yarp::sig::Vector data_tmp = data;
```

Comment:
Given than we are cleaning this part, you should declare the variable as a class instance.